File: py/sanic-pypika-example/src/api/jutsu/jutsu.py
from uuid import UUID
from sanic import Blueprint, Request, json
from sanic.response import empty
from sanic_ext import validate
from ...types.types import JutsuNew, JutsuUpdates

router = Blueprint('jutsu', url_prefix="/api/jutsu")


@router.get("/<id:uuid>")
async def get_jutsu(request: Request, id: UUID):
    """Get jutsu"""
    try:
        jutsu = await request.app.ctx.providers.pgdal.get_jutsu(id)
        return json(jutsu, status=200)
    except Exception:
        return empty(status=500)


@router.post("/")
@validate(json=JutsuNew)
async def create_jutsu(request: Request):
    """Insert new jutsu"""
    try:
        jutsu_new: JutsuNew = request.json
        jutsu = await request.app.ctx.providers.pgdal.create_jutsu(jutsu_new)
        return json(jutsu, status=200)
    except Exception:
        return empty(status=500)


@router.put("/<id:uuid>")
@validate(json=JutsuUpdates)
async def update_jutsu(request: Request, id: UUID,):
    """Update existing jutsu"""
    try:
        jutsu_updates: JutsuUpdates = request.json
        jutsu = await request.app.ctx.providers.pgdal.update_jutsu(id, jutsu_updates)
        return json(jutsu, status=200)
    except Exception:
        return empty(status=500)


@router.delete("/<id:uuid>")
async def delete_jutsu(request: Request, id: UUID,):
    """Delete existing jutsu"""
    try:
        jutsu = await request.app.ctx.providers.pgdal.delete_jutsu(id)
        return json(jutsu, status=200)
    except Exception:
        return empty(status=500)

File: go/chi-squirrel-example/api/jutsu/jutsu.go
package jutsu

import (
	"chi-squirrel-example/api/utils"
	"chi-squirrel-example/providers"
	"chi-squirrel-example/types"
	"net/http"

	"github.com/go-chi/chi/v5"
)

func RegisterJutsu(app chi.Router, providers *providers.Providers) {
	app.Route("/api/jutsu", func(r chi.Router) {
		// Get jutsu
		r.Get("/{id}/", func(w http.ResponseWriter, r *http.Request) {
			id := chi.URLParam(r, "id")
			jutsu, err := providers.PGDAL.GetJutsu(id)
			if err != nil {
				utils.SendStatus(w, http.StatusInternalServerError)
				return
			}
			utils.SendJSON(w, http.StatusOK, jutsu)
		})

		// Insert jutsu
		r.Post("/", func(w http.ResponseWriter, r *http.Request) {
			var jutsuNew types.JutsuNew
			err := utils.ReadJSON(r, &jutsuNew)
			if err != nil {
				utils.SendStatus(w, http.StatusBadRequest)
				return
			}
			jutsu, err := providers.PGDAL.CreateJutsu(jutsuNew)
			if err != nil {
				utils.SendStatus(w, http.StatusInternalServerError)
				return
			}
			utils.SendJSON(w, http.StatusOK, jutsu)
		})

		// Update jutsu
		r.Put("/{id}/", func(w http.ResponseWriter, r *http.Request) {
			id := chi.URLParam(r, "id")
			var jutsuUpdates types.JutsuNew
			err := utils.ReadJSON(r, &jutsuUpdates)
			if err != nil {
				utils.SendStatus(w, http.StatusBadRequest)
				return
			}
			jutsu, err := providers.PGDAL.UpdateJutsu(id, jutsuUpdates)
			if err != nil {
				utils.SendStatus(w, http.StatusInternalServerError)
				return
			}
			utils.SendJSON(w, http.StatusOK, jutsu)
		})

		// Delete Jutsu
		r.Delete("/{id}/", func(w http.ResponseWriter, r *http.Request) {
			id := chi.URLParam(r, "id")
			jutsu, err := providers.PGDAL.DeleteJutsu(id)
			if err != nil {
				utils.SendStatus(w, http.StatusInternalServerError)
				return
			}
			utils.SendJSON(w, http.StatusOK, jutsu)
		})
	})
}
